packages/core/src/types.ts

```diff
@@ -371,10 +371,10 @@ export type TransitionConfigOrTarget<
   TransitionConfigTarget<TContext, TEvent> | TransitionConfig<TContext, TEvent>
 >;
 
-type TransitionsConfigMap<TContext, TEvent extends EventObject> = {
+export type TransitionsConfigMap<TContext, TEvent extends EventObject> = {
   [K in TEvent['type']]?: TransitionConfigOrTarget<
     TContext,
-    TEvent extends { type: K } ? TEvent : never
+    TEvent & { type: K }
```

Comment:
The previous version results in more readable IDE tooltips:

**Current**
<img width="802" alt="Screenshot 2021-01-15 at 14 43 02" src="https://user-images.githubusercontent.com/9800850/104734113-05194e00-5740-11eb-88ab-26ea9677b91c.png">

**Previous**
<img width="845" alt="Screenshot 2021-01-15 at 14 41 21" src="https://user-images.githubusercontent.com/9800850/104734019-e1ee9e80-573f-11eb-8c86-4067f6219fbc.png">

Reply:
Fixed